src/stages/appflow-ingestion.ts

```diff
@@ -0,0 +1,136 @@
+  readonly flowTasks?: appflow.CfnFlow.TaskProperty[];
+}
+
+export class AppFlowIngestionStage extends StateMachineStage {
+  readonly targets?: events.IRuleTarget[];
+  readonly eventPattern?: events.EventPattern;
+  readonly stateMachine: sfn.StateMachine;
+  readonly flowObject: sfn.CustomState;
+
+  constructor(scope: Construct, id: string, props: AppFlowIngestionStageProps) {
+    super(scope, id, props);
+
+    const { flowName, flowExecutionStatusCheckPeriod, destinationFlowConfig, sourceFlowConfig, flowTasks } = props;
+
+    if (!flowName) {
+      // Check required props for CfnFlow create and except if not provided
+      if (!destinationFlowConfig || !sourceFlowConfig || !flowTasks) {
+        throw new Error(
+          "if 'flowName' is not specified, 'destinationFlowConfig', 'sourceFlowConfig' & 'tasks' are required properties",
+        );
+      }
+      const flow = new appflow.CfnFlow(this, `${id}-flow`, {
+        destinationFlowConfigList: [destinationFlowConfig],
+        flowName: `${id}-flow`,
+        sourceFlowConfig: sourceFlowConfig,
+        tasks: flowTasks,
+        triggerConfig: {
+          triggerType: "OnDemand",
+        },
+      });
+      this.flowObject = this.createStartFlowCustomTask(flow.flowName);
+    } else {
+      this.flowObject = this.createStartFlowCustomTask(flowName);
+    }
+
+    // Create check flow execution status step function task
+    const flowExecutionRecords = this.createCheckFlowExecutionTask(id);
+    // Create step function loop to check flow execution status
+    const flowObjectExecutionStatus = new sfn.Choice(this, "check-flow-execution-status");
+    const flowObjectExecutionStatusWait = new sfn.Wait(this, "wait-before-checking-flow-status-again", {
+      time: sfn.WaitTime.duration(flowExecutionStatusCheckPeriod ?? Duration.seconds(15)),
+    });
+
+    const definition = this.flowObject
+      .next(flowObjectExecutionStatusWait)
+      .next(flowExecutionRecords)
+      .next(
+        flowObjectExecutionStatus
+          .when(sfn.Condition.stringEquals("$.FlowExecutionStatus", "Successful"), new sfn.Succeed(this, "success"))
+          .when(
+            sfn.Condition.stringEquals("$.FlowExecutionStatus", "Error"),
+            new sfn.Fail(this, "failure", {
+              error: "WorkflowFailure",
+              cause: "AppFlow failure",
+            }),
+          )
+          .otherwise(flowObjectExecutionStatusWait),
+      );
+
+    ({
+      eventPattern: this.eventPattern,
+      targets: this.targets,
+      stateMachine: this.stateMachine,
+    } = this.createStateMachine(definition, props));
+    this.stateMachine.addToRolePolicy(
+      new iam.PolicyStatement({
+        effect: iam.Effect.ALLOW,
+        actions: ["appflow:StartFlow"],
+        resources: ["*"],
+      }),
+    );
+  }
+
+  private createStartFlowCustomTask(flowName: string): sfn.CustomState {
+    const stateJson = {
```

Comment:
We could probably make this cleaner by using the [CallAwsService](https://docs.aws.amazon.com/cdk/api/v1/docs/@aws-cdk_aws-stepfunctions-tasks.CallAwsService.html) task. But we can save that for a separate PR.

Reply:
Will handle in #223